src/util/graph.h

```diff
@@ -101,7 +102,19 @@ void intersection(
   }
 }
 
-// a generic graph class with a parametric node type
+/// A generic directed graph with a parametric node type.
+///
+/// The nodes of the graph are stored in the only field of the class, a
+/// std::vector named `nodes`. Nodes are instances of class graph_nodet<E> or a
+/// subclass of it. Graph edges contain a payload object of parametric type E.
+/// By default E is instantiated with an empty class (empty_edget).
+///
+/// Each node is identified by its offset inside the `nodes` vector. The
+/// incomding and outgoing edges of a node are stored as std::maps in the fields
+/// `in` and `out` of the graph_nodet<E>. These maps associate a node identifier
+/// (the offset) to an edge payload (of type E). In fact, observe that two
+/// instances of E are stored per a "mathematical" edge from node A to node B,
```

Comment:
As with any academic writing, the use of quotes is alarming. You might consider the fact that you had to add two paragraphs of text to explain one word sufficient feedback that the wording needs review. I don't think "formal" would help much - you might just have to go all the way and merge in most of the paragraphs you pasted here as comment?